### PR TITLE
add formatter-action

### PR DESCRIPTION
**Pull request by kumagi**

### CODE REVIEW

**Comment by tomokinat** [comment]
Previous CI failure seems due to this change https://github.com/llvm/llvm-project/commit/0f4d5f85191c124d13046ceabc04d5861ae7ab6a, so I'd recommend to re-write .clang-format to unify the behavior of various versions.

**Comment by kumagi** [comment]
OK, but that change should be done in another PR.

File: .github/workflows/formatter.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+on: push
+name: Code Format Checker
+jobs:
+  lint:
+    name: Code Format Checker
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checker
+      uses: kumagi/clang-format-action@check-diff
```
Comment (tomokinat) [inline .github/workflows/formatter.yml]: Can you include the version of clang-format in the tag?

File: .github/workflows/formatter.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tomokinat) [inline .github/workflows/formatter.yml]: Also I think the token is no longer needed as the action now just clones the repo.

File: .github/workflows/formatter.yml (hunk 1 above)
Comment (kumagi) [reply .github/workflows/formatter.yml]: Done.

File: .github/workflows/formatter.yml (hunk 2 above)
Comment (kumagi) [reply .github/workflows/formatter.yml]: Removed.

**Pull request merged**